src/ansys/dynamicreporting/core/utils/report_remote_server.py

```diff
@@ -1033,14 +1047,118 @@ def load_templates(self, templates_json):
                 break
 
         root_attr = templates_json[root_id_str]
-        root_template = self._populate_template(root_attr, None)
+        root_template = self._populate_template(root_id_str, root_attr, None)
         self.put_objects(root_template)
         id_template_map = {}
         id_template_map[root_id_str] = root_template
         self._build_templates_from_parent(root_id_str, id_template_map, templates_json)
         self._update_changes(root_id_str, id_template_map, templates_json)
 
 
+def _check_template(template_id_str, template_attr):
+    # Check template_id_str
+    if not _check_template_name_convention(template_id_str):
+        raise TemplateEditorJSONLoadingError(
+            f"The loaded JSON file has an invalid template name: '{template_id_str}' as the key.\n"
+            "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+        )
+
+    # Check parent and children template name convention
+    if not _check_template_name_convention(template_attr["parent"]):
+        raise TemplateEditorJSONLoadingError(
+            f"The loaded JSON file has an invalid template name: '{template_attr['parent']}' "
+            f"that does not have the correct name convection under the key: 'parent' of '{template_id_str}'\n"
+            "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+        )
+
+    for child_name in template_attr["children"]:
+        if not _check_template_name_convention(child_name):
+            raise TemplateEditorJSONLoadingError(
+                f"The loaded JSON file has an invalid template name: '{child_name}' "
+                f"that does not have the correct name convection under the key: 'children' of '{template_id_str}'\n"
+                "Please note that the naming convention is 'Template_{NONE_NEGATIVE_NUMBER}'"
+            )
+
+    # Check key names
+    agreed_keys = {
+        "name",
+        "report_type",
+        "date",
+        "tags",
+        "params",
+        "sort_selection",
+        "item_filter",
+        "parent",
+        "children",
+    }
+    for key in template_attr.keys():
+        if key not in agreed_keys:
+            raise TemplateEditorJSONLoadingError(
+                f"The loaded JSON file is using an unknown key: '{key}' other than any in the JSON schema.\n"
+                f"Please check the '{key}' entry under '{template_id_str}' in your JSON file as you might have a typo in that entry."
+            )
+
+    # Check missing keys
+    if len(template_attr) != len(agreed_keys):  # Missing one or more entries
+        keys = template_attr.keys()
+        missing_keys = list(agreed_keys - set(keys))
+        raise TemplateEditorJSONLoadingError(
+            f"The loaded JSON file is missing keys: {missing_keys}.\n"
+            f"Please check them under '{template_id_str}' in your JSON file for the missing keys."
+        )
+
+    # Check report_type
+    report_types = {
```

Comment:
@zhang-yuanrui 
1. The generator layouts are missing - they should be included.
2. This is a bad idea, as it assumes that every time someone adds a template type, they will have to remember to add it also here. Which will not happen more often than not, leading to broken code.
We need to find a way to automatically generate this list.

Reply:
I agree with Marina. Again, this is very similar to what you did previously and must be avoided. You are basically creating another point of failure should someone add a new report type.  Please do not hardcode anything.